pkg/resourceapply/helpers.go

```diff
@@ -244,7 +244,7 @@ func ApplyGenericWithHandlers[T kubeinterfaces.ObjectInterface](
 	required T,
 	options ApplyOptions,
 	projectFunc func(required *T, existing T),
-	getRecreateReasonFunc func(required T, existing T) string,
+	getRecreateReasonFunc func(required T, existing T) (string, metav1.DeletionPropagation, error),
```

Comment:
i think this would make the changes and UX nicer
```suggestion
	getRecreateReasonFunc func(required T, existing T) (string, *metav1.DeletionPropagation, error),
```

Reply:
done